packages/nodes-base/nodes/Schedule/ScheduleTrigger.node.ts

```diff
@@ -430,7 +431,24 @@ export class ScheduleTrigger implements INodeType {
 				Second: moment.tz(timezone).format('ss'),
 				Timezone: moment.tz(timezone).format('z Z'),
 			};
-			this.emit([this.helpers.returnJsonArray([resultData])]);
+			const weeklyExecution = i ? (staticData[i] as IDataObject) : ({} as IDataObject);
```

Comment:
We should explicitly check if `i === undefined` since `i = 0` would fall into the "else" part of the ternary operator